lib/ansible/playbook/included_file.py

```diff
@@ -195,11 +196,13 @@ def process_include_results(results, iterator, loader, variable_manager):
                             # use idx + pos due to relative indexing
                             inc_file = included_files[idx + pos]
                         except ValueError:
+                            orig_inc_file._results.append(res)
                             included_files.append(orig_inc_file)
                             inc_file = orig_inc_file
 
                         try:
                             inc_file.add_host(original_host)
+                            inc_file._results.append(res)
```

Comment:
I have to say, I don't remember why I originally added this line here, and whether it's actually needed.  It seems like it will duplicate the `res` in `_results`.  😵

Reply:
That appears to be correct however I think this is the right place to add the result. We should instead remove the append on line 199 to avoid duplication and to append results for both existing and new `IncludedFile`, so on top of this PR:
```diff
diff --git a/lib/ansible/playbook/included_file.py b/lib/ansible/playbook/included_file.py
index 57f5b869d9d..c256d70b740 100644
--- a/lib/ansible/playbook/included_file.py
+++ b/lib/ansible/playbook/included_file.py
@@ -208,7 +208,6 @@ class IncludedFile:
                             # use idx + pos due to relative indexing
                             inc_file = included_files[idx + pos]
                         except ValueError:
-                            orig_inc_file._results.append(res)
                             included_files.append(orig_inc_file)
                             inc_file = orig_inc_file
```

🤯

Reply:
I've re-read the code like 12 times this morning, and I think that is correct.  🤯++